### PR TITLE
populate defaultValues in webidl dict. converter when passing null or undefined

### PR DESCRIPTION
**Pull request by KhafraDev**
… undefined

The test explains the commit purpose better.

`benchmarks/fetch/request-creation.mjs` was consistently 10-15% faster. Not sure how @Uzlopak @tsctx, was this a fluke?

### CODE REVIEW

File: lib/web/fetch/webidl.js [hunk 1]
```diff
@@ -354,12 +354,9 @@ webidl.interfaceConverter = function (i) {
 
 webidl.dictionaryConverter = function (converters) {
   return (dictionary, prefix, argument) => {
-    const type = webidl.util.Type(dictionary)
     const dict = {}
 
-    if (type === 'Null' || type === 'Undefined') {
-      return dict
-    } else if (type !== 'Object') {
+    if (dictionary != null && typeof dictionary !== 'object') {
```
Comment (tsctx) [inline lib/web/fetch/webidl.js:359]:
```suggestion
    if (dictionary != null && webidl.util.Type(dictionary) !== 'Object') {
```

The 'function' is also an 'Object' but it errors.

File: lib/web/fetch/webidl.js [hunk 2]
```diff
@@ -378,13 +375,13 @@ webidl.dictionaryConverter = function (converters) {
...
+      let value = dictionary?.[key]
+      const hasDefault = defaultValue !== undefined
 
       // Only use defaultValue if value is undefined and
       // a defaultValue options was provided.
-      if (hasDefault && value !== null) {
-        value ??= defaultValue()
+      if (hasDefault && value === undefined) {
+        value = defaultValue()
       }
```
Comment (tsctx) [inline lib/web/fetch/webidl.js:385]:
```suggestion
      if (hasDefault && value === undefined) {
        value = defaultValue()
      } else if (required === true && value === undefined) {
        throw webidl.errors.exception({
            header: prefix,
            message: `Missing required key "${key}".`
          })
      }
```

**Comment by KhafraDev** [comment]
I'm still confused how Request creation (with no RequestInit) is 10-15% faster.

**Review by tsctx — APPROVED** [review]
The constructor of Request has an empty object by default, so passing undefined to RequestInit will result in default value.
The performance increase is probably due to the removal of Object.hasOwn.

**Comment by KhafraDev** [comment]
I thought it could be Object.hasOwn, but the extra "overhead" from now having to loop over all of the converters made me think it would be slower. That's why I bothered benchmarking it in the first place.

**Pull request merged**